ros/processors/report_processor.py

```diff
@@ -0,0 +1,112 @@
+        consumer.subscribe(['platform.upload.resource-optimization'])
+        while running:
+            msg = consumer.poll(1.0)
+            if msg is None:
+                continue
+            if msg.error():
+                print(msg.error())
+                continue
+            try:
+                self.msg = json.loads(msg.value().decode("utf-8"))
+                self.handle_message()
+            except json.decoder.JSONDecodeError:
+                print(f"Unable to decode kafka message - {msg.value()}")
+            except Exception as err:
+                print(f"An error occurred during message processing - {err}")
+            finally:
+                consumer.commit()
+
+        consumer.close()
+
+    def handle_message(self):
+        self.report_url = self.msg.get('url', None)
+        metadata = self.msg.get('metadata', None)
+        if not metadata:
+            return None
+        insights_id = metadata['insights_id']
+        if not insights_id:
+            return None
+        rh_identity = self.msg.get('b64_identity', None)
+        host = fetch_host_from_inventory(insights_id, rh_identity)
+        if not host['results']:
+            print("No record found. Make sure system is registered in insights")
+            return None
+        host_id = host['results'][0]['id']
+        if not self.report_url:
+            print("kafka message missing report url")
+        report_tar_gz = self._download_report()
+        performance_record = self._extract_performance_record(report_tar_gz)
+
+        with app.app_context():
+            if performance_record:
+                performance_score = self._calculate_performance_score(performance_record)
+                profile = PerformanceProfile.query.filter_by(inventory_id=host_id).first()
+                if profile:
+                    profile.performance_record = performance_record
+                    profile.performance_score = performance_score
+                else:
+                    record = PerformanceProfile(inventory_id=host_id,
+                                                performance_record=performance_record,
+                                                performance_score=performance_score)
+                    db.session.add(record)
+
+                db.session.commit()
+
+    def _calculate_performance_score(self, performance_record):
+        memory_score = (float(performance_record['avg_memory_used']) / float(performance_record['avg_memory'])) * 100
+        performance_score = {'memory_score': int(memory_score)}
+        return performance_score
+
+    def _download_report(self):
+        download_response = requests.get(self.report_url)
+        if download_response.status_code != HTTPStatus.OK:
+            print("Unable to download the report")
+        return download_response.content
+
+    def _extract_performance_record(self, report_tar_gz):
+        tar = tarfile.open(fileobj=BytesIO(report_tar_gz), mode='r:gz')
```

Comment:
```suggestion
        tar = tarfile.open(fileobj=BytesIO(report_tar_gz), mode='r:*')
```

On `yupana` service, recently we added support for `xz & bz2` along with `gz`, I think it would be good to have same support in ROS too. What do you think?

Accordingly, you will need to change variable naming.

Reply:
:+1: